imgee/views/index.py

```diff
@@ -26,215 +20,52 @@ def index():
-        url = request.referrer
-    else:
-        url = url_for('profile_view', profile=profile_name)
-    return url
-
-def stored_file_data(stored_file):
-    return dict(
-        title=stored_file.title,
-        uploaded=stored_file.created_at.strftime('%B %d, %Y'),
-        filesize=app.jinja_env.filters['filesizeformat'](stored_file.size),
-        imgsize='%s x %s' % (stored_file.width, stored_file.height),
-        url=url_for('view_image', profile=stored_file.profile.name, image=stored_file.name),
-        thumb_url=url_for('get_image', image=stored_file.name, size=app.config.get('THUMBNAIL_SIZE')))
-
-def generate_thumbs(image):
-    def generate_thumb(image, size):
-        try:
-            utils.get_image_url(image, size)
-        except async.StillProcessingException:
-            pass
-    generate_thumb(image, '75x75')
-    sizes = [250, 400, 430, 600, 770]
-    for size in sizes:
-        if size > image.width:
-            break
-        else:
-            generate_thumb(image, str(size))
-
-@app.route('/<profile>/new', methods=['GET', 'POST'])
-@load_model(Profile, {'name': 'profile'}, 'profile',
-    permission=['new-file', 'siteadmin'], addlperms=lastuser.permissions)
-def upload_file(profile):
-    upload_form = forms.UploadImageForm()
-    if upload_form.validate_on_submit():
-        file_ = request.files['file']
-        title, job, stored_file = save(file_, profile=profile)
-        generate_thumbs(stored_file)
-        flash('"%s" uploaded successfully.' % title)
-        return redirect(_redirect_url_frm_upload(profile.name))
-    return render_template('form.html', form=upload_form, profile=profile)
-
-@app.route('/<profile>/new.json', methods=['POST'])
-@load_model(Profile, {'name': 'profile'}, 'profile',
-    permission=['new-file', 'siteadmin'], addlperms=lastuser.permissions)
-def upload_file_json(profile):
-    upload_form = forms.UploadImageForm()
-    if upload_form.validate_on_submit():
-        file_ = request.files['file']
-        title, job, stored_file = save(file_, profile=profile)
-        generate_thumbs(stored_file)
-        update_form = forms.UpdateTitle()
-        update_form.title.data = stored_file.title
-        form = render_template('edit_title_form.html', form=update_form, formid='edit_title_' + stored_file.name)
-        return jsonify(
-            status=True, message="%s uploaded successfully" % title, form=form,
-            update_url=url_for('update_title_json', profile=profile.name,file=stored_file.name),
-            image_data=stored_file_data(stored_file))
-    else:
-        response = jsonify(status=False, message=upload_form.errors['file'])
-        response.status_code = 403
-        return response
-
 @app.route('/<profile>/popup')
+@lastuser.requires_login
 @load_model(Profile, {'name': 'profile'}, 'profile',
-    permission=['view', 'siteadmin'], addlperms=lastuser.permissions)
+    permission=['view', 'siteadmin'])
```

Comment:
You need the `addlperms` parameter to receive `siteadmin`. It's needed anywhere you're checking for the `siteadmin` permission.